synthtool/gcp/templates/python_mono_repo_library/noxfile.py.j2

```diff
@@ -527,3 +527,64 @@ def prerelease_deps(session, protobuf_implementation):
             "PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION": protobuf_implementation,
         },
     )
+
+
+@nox.session(python="{{ unit_test_python_versions | last }}")
+@nox.parametrize(
+    "protobuf_implementation",
+    ["python", "upb"],
+)
+def core_deps_from_source(session, protobuf_implementation):
+    """Run all tests with local versions of core dependencies installed."""
+
+    # Install all dependencies
+    session.install(".[all, tests, tracing]")
```

Comment:
I'm not familiar with the syntax in the string. Do you have a reference?

Also, could you add a comment to distinguish between the purposes of this install vs `unit_deps_all` vs `system_deps_all` vs `constraints_deps` below?

Reply:
This is the syntax for [installing extras](https://packaging.python.org/en/latest/tutorials/installing-packages/#installing-extras). We don't have any extras in `google-cloud-python`, so I've removed the extras but they do exist in split repositories. For example: https://github.com/search?q=org%3Agoogleapis%20path%3Asetup.py%20tracing&type=code

Fixed in https://github.com/googleapis/synthtool/pull/2061/commits/a07785120b220e6f5fcbbe31e9e4fbea7eb4f3d1. I'll push another commit to add comments